plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -255,6 +271,32 @@ private Object toIntegerVarcharMapBlock(Map<Integer, String> values)
             blockBuilder.closeEntry();
             return integerToVarcharMapType.getObject(blockBuilder, 0);
         }
+
+        private Object getIntegerArrayBlock(List<Integer> values)
```

Comment:
Since the type is changed in 1st commit `Change to use TypeManager to get ArrayType signature`. then type-related changes should be part of 1st commit.

Reply:
Make sense. Let me put those changes into the 1st commit.